### PR TITLE
RDCC-208

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RDCC-208

### Change description ###
Updated professional user response to include user identifier, status and roles (please note roles will always return empty until SIDAM connection occurs in the future!)

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUserResponse.java [hunk 1]
```diff
@@ -2,23 +2,33 @@
...
     @JsonProperty
-    private final String lastName;
+    private ProfessionalUserStatus status;
     @JsonProperty
-    private final String email;
+    private List<String> roles = new ArrayList<String>();
 
     public ProfessionalUserResponse(ProfessionalUser user) {
-        this.id = user.getId().toString();
+        this.userIdentifier = user.getUserIdentifier();
```
Comment (satyachundur) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUserResponse.java]: Do you need check for null? which we had the problem on SuperUserResponse.java

Comment (AdilOoze) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUserResponse.java]: Ahh yes, I will include the same piece of code. Once again, good catch!

**Comment by AdilOoze** [comment]
> 👍 I hope this class has already unit tests and integration tests done

Yes, they were created by Tabby as part of RDCC-78.
This is simply updating the response to include a few new fields.
The tests pass in my local machine!

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUserResponse.java [hunk 2]
```diff
@@ -2,23 +2,34 @@
...
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.util.StringUtils;
 import uk.gov.hmcts.reform.professionalapi.domain.ProfessionalUser;
+import uk.gov.hmcts.reform.professionalapi.domain.ProfessionalUserStatus;
 
 public class ProfessionalUserResponse {
```
Comment (shreedhar-hmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUserResponse.java]: Please use ProfessionalUsersResponse directly as this has been used already in other endpoint. Remove/Delete this java file.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java [hunk 3]
```diff
@@ -76,6 +78,18 @@ public ProfessionalUser findProfessionalUserByEmailAddress(String email) {
         return user;
     }
 
+    public ProfessionalUser findProfessionalUserByEmailAddressForActiveOrganisation(String email) {
```
Comment (akiohmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java]: We should not be having two duplicate service methods with different names. Please remove this and if you need to validation add it in the original method if possible, add an extra param if need be but never two analogous methods in the same interface.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/service/ProfessionalUserService.java [hunk 4]
```diff
@@ -13,5 +13,7 @@
     ProfessionalUser findProfessionalUserByEmailAddress(String email);
 
     List<ProfessionalUser> findProfessionalUsersByOrganisation(Organisation existingOrganisation, boolean showDeleted);
+
+    ProfessionalUser findProfessionalUserByEmailAddressForActiveOrganisation(String email);
```
Comment (akiohmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/service/ProfessionalUserService.java]: Please remove as per below comment.

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUsersResponse.java [hunk 5]
```diff
@@ -18,7 +18,7 @@
 public class ProfessionalUsersResponse {
 
     @JsonProperty
-    private UUID userIdentifier;
+    private String userIdentifier;
```
Comment (akiohmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUsersResponse.java]: Better to leave it as a UUID and remove the toString part below

File: src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUsersResponse.java [hunk 6]
```diff
@@ -31,7 +31,8 @@
     private List<String> roles = new ArrayList<String>();
 
     public ProfessionalUsersResponse(ProfessionalUser user) {
-        this.userIdentifier = user.getUserIdentifier();
+        this.userIdentifier = StringUtils.isEmpty(user.getUserIdentifier())
```
Comment (akiohmcts) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/controller/response/ProfessionalUsersResponse.java]: Better to use Java Optionals e.g. Optional<UUID> userIdOpt = new Optional(user.getUserIdentifier);

**Pull request merged**